apps/wallet/src/background/accounts/zklogin/ZkLoginAccount.ts

```diff
@@ -282,12 +342,63 @@ export class ZkLoginAccount
 			randomness: randomness.toString(),
 			jwt,
 		};
-		ephemeralValue[serializeNetwork(activeNetwork)] = credentialsData;
-		await this.setEphemeralValue(ephemeralValue);
+		await this.saveCredentialsData(credentialsData);
 		await this.onUnlocked();
+
+		// On re-auth, we check if the account for the complementary
+		// legacy/non-legacy address needs to be imported. Additionally, if the
+		// complementary account has been imported before and is unlocked, we
+		// update its credentials. This sync does not need to block the login
+		// process.
+		this.#syncAlternateAccount(jwt, credentialsData);
+
 		return credentialsData;
 	}
 
+	async #syncAlternateAccount(jwt: string, credentialsData: CredentialData) {
+		const currentAddress = await this.address;
```

Comment:
This check seems really weird, why is it using the current address? You should be able to do this based on just the stored accounts not based on the current state